testsuite/integration/xts/src/test/java/org/jboss/as/test/xts/annotation/client/TransactionalTestCase.java

```diff
@@ -44,17 +49,33 @@
 public class TransactionalTestCase {
 
     private static final String DEPLOYMENT_NAME = "transactional-test";
-
-    @ArquillianResource
-    private ManagementClient managementClient;
+    private static final String SERVER_HOST_PORT = TestSuiteEnvironment.getServerAddress() + ":" + TestSuiteEnvironment.getHttpPort();
+    private static final String DEPLOYMENT_URL = "http://" + SERVER_HOST_PORT + "/" + DEPLOYMENT_NAME;
 
     @Deployment
     public static WebArchive getDeployment() {
+        final String javaHome = getSystemProperty("java.home");
         final WebArchive webArchive = ShrinkWrap.create(WebArchive.class, DEPLOYMENT_NAME + ".war")
                 .addClass(TransactionalClient.class)
                 .addClass(TransactionalService.class)
                 .addClass(TransactionalServiceImpl.class)
+                .addClass(TestSuiteEnvironment.class)
                 .addAsManifestResource(new StringAsset("Dependencies: org.jboss.xts,org.jboss.jts\n"), "MANIFEST.MF")
+                .addAsManifestResource(createPermissionsXmlAsset(
+                        // TransactionalClient#newInstance
+                        new RuntimePermission("getClassLoader"),
+                        new RuntimePermission("accessDeclaredMembers"),
+                        new FilePermission(javaHome + File.separator + "lib" + File.separator + "jaxws.properties", "read"), // javax.xml.ws.spi.FactoryFinder
+                        new FilePermission(javaHome + File.separator + "lib" + File.separator + "wsdl.properties", "read"),
+                        new PropertyPermission("user.dir", "read"),
+                        new RuntimePermission("org.apache.cxf.permission"),
+                        new SocketPermission(SERVER_HOST_PORT, "connect,resolve"),
+                        new RuntimePermission("createClassLoader"),
```

Comment:
This does not look right. I tried running this locally and ran into https://github.com/wildfly/wildfly/pull/8121

After that was fixed I now get:

```
javax.xml.ws.WebServiceException: java.security.AccessControlException: WFSM000001: Permission check failed (permission "("java.lang.RuntimePermission" "getClassLoader")" in code source "(vfs:/content/transactional-test.war/WEB-INF/classes <no signer certificates>)" of "null")
	at org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:273)
	at org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:175)
	at java.lang.ClassLoader.checkClassLoaderPermission(ClassLoader.java:1525)
	at java.lang.Thread.getContextClassLoader(Thread.java:1436)
	at javax.xml.ws.spi.FactoryFinder.find(FactoryFinder.java:70)
	at javax.xml.ws.spi.Provider.provider(Provider.java:110)
	at javax.xml.ws.Service.<init>(Service.java:57)
	at javax.xml.ws.Service.create(Service.java:687)
	at org.jboss.as.test.xts.annotation.client.TransactionalClient.newInstance(TransactionalClient.java:57)
	at org.jboss.as.test.xts.annotation.client.TransactionalTestCase.testActiveTransaction(TransactionalTestCase.java:76)
```

Which is a problem with the javax.xml.ws.Service class (it should be doing its factory finder stuff in a PA). This is a real bug in our code when running under a security manager, and adding these permissions simply covers it up.